### DISCUSSION TITLE
[Question] [Navigation and routing] how to append data to the browser of other computer ?

### DISCUSSION THREAD

**Discussion by jjjunehkim** [flet-dev/flet#3090] [Q&A] [answered]
### Question

I have two computer in same network. 
The program ran on "A" pc(computer)
I launched a chrome browser on each of the two computers. 
and
"A" pc  -> "http://192.168.35.35:8550 
 "B" pc  -> "http://192.168.35.35:8550//data_table
 << 192.168.35.35 -> it is my ip address.>>

When I click a button on "A" computer, I want to add data to a data table on my computer. 
but. it doesn't...
please help me how to

### Code sample

```python
import flet as ft

def main(page: ft.Page):

    data_table_sample=ft.DataTable(
                        width=400,
                        bgcolor="yellow",
                        border=ft.border.all(2, "red"),
                        border_radius=10,
                        vertical_lines=ft.border.BorderSide(3, "blue"),
                        horizontal_lines=ft.border.BorderSide(1, "green"),
                        sort_column_index=0,
                        sort_ascending=True,
                        heading_row_color=ft.colors.BLACK12,
                        heading_row_height=100,
                        data_row_color={"hovered": "0x30FF0000"},
                        show_checkbox_column=True,
                        divider_thickness=0,
                        column_spacing=200,
                        columns=[
                            ft.DataColumn(
                                ft.Text("Column 1"),

                            ),
                            ft.DataColumn(
                                ft.Text("Column 2"),
                                tooltip="This is a second column",
                                numeric=True,

                            ),
                        ],
                        rows=[
                            ft.DataRow([ft.DataCell(ft.Text("A")), ft.DataCell(ft.Text("1"))]),
                        ],
                    )
    


    def add_data_1(e): 
        data_table_sample.rows.append(
            ft.DataRow(
                cells=[
                    ft.DataRow([ft.DataCell(ft.Text("B")), ft.DataCell(ft.Text("2"))])
                ],
            )
        )
        print("appended data_1 !!")
        page.update()


    def add_data_2(e):

        data_table_sample.rows.append(
            ft.DataRow(
                cells=[
                    ft.DataRow([ft.DataCell(ft.Text("C")), ft.DataCell(ft.Text("3"))])
                ],
            )
        )
        print("appended data_2 !!")
        page.update()

    def route_change(route):
        global message
        page.views.clear()
        page.views.append(
            ft.View(
                "/",
                [
                    ft.ElevatedButton(text="add_data", on_click = add_data_1, data = "add_data"),
                ],
            )
        )
        if page.route == "/data_table":
            page.views.append(
                ft.View(
                    "/data_table",
                    [
                        ft.ElevatedButton(text="add_data", on_click = add_data_2, data = "add_data"),
                        data_table_sample
                    ],
                )
            )
        page.update()

    def view_pop(view):
        page.views.pop()
        top_view = page.views[-1]
        page.go(top_view.route)


    page.on_route_change = route_change
    page.on_view_pop = view_pop
    page.go(page.route)


ft.app(target=main, host="192.168.35.35", port= 8550, view=ft.AppView.WEB_BROWSER)

 host="192.168.35.35" --> it is my ip-address
```

### Error message

```bash
n/a
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by abdokhalil07** [comment, marked as answer]
import flet as ft

def main(page: ft.Page):

    data_table_sample=ft.DataTable(width=400,border=ft.border.all(2, "red"),border_radius=10,vertical_lines=ft.border.BorderSide(3, "blue"),horizontal_lines=ft.border.BorderSide(1, "green"),
                        sort_column_index=0,sort_ascending=True,heading_row_color=ft.colors.BLACK12,heading_row_height=100,data_row_color={"hovered": "0x30FF0000"},show_checkbox_column=True,divider_thickness=0,column_spacing=200,
                            columns=[
                            ft.DataColumn(
                                ft.Text("Column 1"),),
                            ft.DataColumn(
                                ft.Text("Column 2"),
                                tooltip="This is a second column",
                                numeric=True,
                            ),
                        ],
                        rows=[
                            ft.DataRow([ft.DataCell(ft.Text("A")), ft.DataCell(ft.Text("1"))]),
                        ],
                    )
    

    # Start from her   --------------------------------------------------------------------------------------------------------------------
    def add_data_1(e): 
        page.pubsub.send_all(e)
        page.update()

    def add_data_12(): 

        row = ft.DataRow(
            cells=[
                ft.DataCell(ft.Text("B")),
                ft.DataCell(ft.Text("2"))
                    ],
        )

        return row
    def on_message(row):
        row = add_data_12()
        data_table_sample.rows.append(row)
        page.update()


    page.pubsub.subscribe(on_message)

    def add_data_2(e):
        data_table_sample.rows.append(
            ft.DataRow(
                cells=[
                    ft.DataCell(ft.Text("C")),
                    ft.DataCell(ft.Text("3"))
                ],
            )
        )
        print("appended data_2 !!")
        page.update()
    # End  ------------------------------------------------

    
    def route_change(route):
        global message
        page.views.clear()
        page.views.append(
            ft.View(
                "/",
                [
                    ft.ElevatedButton(text="add_data", on_click = add_data_1, data = "add_data"),
                ],
            )
        )
        if page.route == "/data_table":
            page.views.append(
                ft.View(
                    "/data_table",
                    [
                        ft.ElevatedButton(text="add_data", on_click = add_data_2, data = "add_data"),
                        data_table_sample
                    ],
                )
            )
        page.update()
    
    def view_pop(view):
        page.views.pop()
        top_view = page.views[-1]
        page.go(top_view.route)


    page.on_route_change = route_change
    page.on_view_pop = view_pop
    page.go(page.route)


ft.app(target=main, host="10.0.0.45", port= 8550, view=ft.AppView.WEB_BROWSER)

 #host="192.168.35.35" --> it is my ip-address

**Reply by abdokhalil07** [reply]
try this code.....
try to use https://flet.dev/docs/cookbook/pub-sub

**Reply by jjjunehkim** [reply]
Oh!!  Thank you so so so so.... much!!!
You are really like a angel for me!!
